core/src/main/java/org/infinispan/partitionhandling/impl/PreferConsistencyStrategy.java

```diff
@@ -179,10 +179,8 @@ public void onPartitionMerge(AvailabilityStrategyContext context,
          actualMembers.retainAll(mergedTopology.getMembers());
          mergedAvailabilityMode = AvailabilityMode.DEGRADED_MODE;
       } else {
-         log.debugf("No current topology, recovered only joiners for cache %s", context.getCacheName());
-         mergedTopology = null;
-         actualMembers = null;
-         mergedAvailabilityMode = AvailabilityMode.AVAILABLE;
+         log.debugf("No current topology, recovered only joiners for cache %s. Skipping availability update.", context.getCacheName());
+         return;
```

Comment:
The `doMergePartitions` method name is misleading, it could be just that the old coordinator crashed and the new coordinator just needs to recover one partition. 

I think the idea was to create an initial topology if the cache has no "full" members yet, but there is one node who sent a join request to the old coordinator and didn't get a response with the initial cache topology before sending its status response. The joiner should be retrying to join with the new coordinator though, so I'm not sure it's worth keeping track of these joiners during the recovery process.

The case I was thinking about before is when there is a merge, but a node also left the cluster in the same view. If that node was the last member of one cache, we should make sure that cache's topology is set to `null`. Unfortunately you're right, we were trying to create an initial topology here too, and that probably caused a `NPE` sooner or later...